tests/rptest/tests/upgrade_test.py

```diff
@@ -244,12 +244,19 @@ def test_upgrade_with_all_workloads(self, single_upgrade):
             elif current_version[0:2] == (22, 1) and paused:
                 self._producer.start(clean=False)
 
-        for consumer in self._consumers:
-            consumer.wait()
-
-        assert self._seq_consumer.consumer_status.validator.valid_reads >= wrote_at_least
-        assert self._rand_consumer.consumer_status.validator.total_reads >= self.RANDOM_READ_COUNT * self.RANDOM_READ_PARALLEL
-        assert self._cg_consumer.consumer_status.validator.valid_reads >= wrote_at_least
+            # Wait for consumers to finish work and check invariants.
+            for consumer in self._consumers:
+                consumer.wait()
```

Comment:
LGTM. Wondering if there's any meaningful impact here on the runtime of the test? (not that that would block this change, just curious)

Reply:
Nope. All the data is consumed in under a second anyway.

This is not great for the consumer group consumer because it consumes only once. For the others as well actually, va use they don’t get a good  to exercise more code paths during the upgrade process and assert stuff about monotonicity reliable.

I’ll see how this can improved with a throttled producer and additional assertions ie that producer continues to make progress through upgrades in a later pr which will impact the run time.